src/DynamoCoreWpf/Controls/StartPage.xaml.cs

```diff
@@ -367,29 +374,107 @@ private void RefreshBackupFileList(IEnumerable<string> filePaths)
             IEnumerable<string> filePaths)
         {
             files.Clear();
+
+
+            Stopwatch stopwatch = new Stopwatch();
+
             foreach (var filePath in filePaths.Where(x => x != null))
             {
                 try
                 {
+                    // Skip files which were moved or deleted (consistent with Revit behavior)
+                    if (!System.IO.File.Exists(filePath)) continue;
+
                     var extension = Path.GetExtension(filePath).ToUpper();
                     // If not extension specified and code reach here, this means this is still a valid file
                     // only without file type. Otherwise, simply take extension substring skipping the 'dot'.
                     var subScript = extension.IndexOf(".") == 0 ? extension.Substring(1) : "";
                     var caption = Path.GetFileNameWithoutExtension(filePath);
 
+                    // Measure the performance for this region
+                    // deserializes the file only once
+                    var jsonObject = DeserializeJsonFile(filePath); 
+                    var description = GetGraphDescription(jsonObject);
+                    var thumbnail = GetGraphThumbnail(jsonObject);
+                    var author = GetGraphAuthor(jsonObject);
+
+                    var date = GetDateModified(filePath);
+
                     files.Add(new StartPageListItem(caption)
                     {
                         ContextData = filePath,
                         ToolTip = filePath,
                         SubScript = subScript,
-                        ClickAction = StartPageListItem.Action.FilePath
-                    });
+                        Description = description,
+                        Thumbnail = thumbnail,
+                        Author = author,
+                        DateModified = date,
+                        ClickAction = StartPageListItem.Action.FilePath,
+
+                    }); ;
                 }
                 catch (ArgumentException ex)
                 {
                     DynamoViewModel.Model.Logger.Log("File path is not valid: " + ex.StackTrace);
                 }
             }
+
+            stopwatch.Stop(); // Stop the stopwatch
+
+            Debug.WriteLine($"Time taken to load {filePaths.Count()} files: {stopwatch.ElapsedMilliseconds} ms");
```

Comment:
@dnenov Do you recall the time here, what would be the range for a default 10 paths setup?

Reply:
Oh, yes! 0ms, and I increased the number of files to 50. I tested it with 20 (that's how many recent files I got to) and there is the downtime is 0ms. 

Should I remove this bit now, @QilongTang ?

Reply:
Please do